src/main/scala/mrpowers/jodie/DeltaHelpers.scala

```diff
@@ -12,21 +14,60 @@ object DeltaHelpers {
   /**
    * Gets the latest version of a Delta lake
    */
-  def latestVersion(path: String): Long = {
-    DeltaTable
-      .forPath(SparkSession.active, path)
-      .history(1)
-      .select("version")
-      .head()(0)
-      .asInstanceOf[Long]
+  def latestVersion(path: String): Long =
+    DeltaLog.forTable(SparkSession.active, path).snapshot.version
+
+  def deltaFileSizesDistribution(path: String, condition: Option[String] = None): DataFrame =
+    deltaFileStats(path, condition).select(sizeColumn).summary().toDF(sizeDFColumns: _*)
+
+  def deltaNumRecordsDistribution(path: String, condition: Option[String] = None): DataFrame =
+    deltaFileStats(path, condition).select(numRecordsColumn).summary().toDF(numRecordsDFColumns: _*)
+
+  def deltaPartitionWiseFileSizeDistribution(path: String): DataFrame =
+    getAllPartitionStats(path, statsPartitionColumn, sizeColumn)
+      .toDF(sizeDFColumns: _*)
+
+  def deltaPartitionWiseNumRecordsDistribution(path: String): DataFrame =
+    getAllPartitionStats(path, statsPartitionColumn, numRecordsColumn)
+      .toDF(numRecordsDFColumns: _*)
+
+  def getAllPartitionStats(path: String, groupByCol: String, aggCol: String) = {
+    deltaFileStats(path)
+      .withColumn("norm",when(col(aggCol).equalTo(sizeColumn), col(aggCol).divide(1024*1024)).otherwise(col(aggCol)))
```

Comment:
I think your issue is related to the use of "col" instead of "lit" function, you should change it to something like this:

 `.withColumn("norm",when(lit(aggCol).equalTo(lit(sizeColumn)),col(aggCol).divide(1024*1024)).otherwise(col(aggCol)))`

Reply:
this one worked, thanks @brayanjuls  I have added a separate method to keep the functionality separate